### ISSUE TITLE
[a11y]: [Windows High Contrast Mode]: [Multiselect] The component miss some borders when values are selected

### ISSUE THREAD

**Issue by xrissot-ibm** [carbon-design-system/carbon#19022]
### Package

@carbon/react

### Browser

Chrome

### Operating System

Windows

### Package version

v1.79

### React version

v1.79

### Automated testing tool and ruleset

Manual testing for Review recommandations section

### Description

The component should have borders all around it

### Reproduction/example

use carbon-react playground

### Steps to reproduce

Open Multiselect playground,
Select some values
Activate Windows HCM, it miss some borders around the component:

![Image](https://github.com/user-attachments/assets/7e404772-bea3-40cb-9cf7-0f04cc37f4f5)

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
